_posts/2018-03-11-user-testing-warehouse.md

```diff
@@ -0,0 +1,54 @@
+
+Recurring problems are then raised as design issues for the team to fix.
+
+## User Testing Warehouse
+
+It [had always been my intention](http://localhost:4000/2015/12/31/designing-warehouse-an-overview.html) to run user tests on the [Warehouse project](https://pypi.org/), but it wasn't until the project [received MOSS funding](http://pyfound.blogspot.co.uk/2017/11/the-psf-awarded-moss-grant-pypi.html) that I was able to dedicate the amount of time required to set up and conduct these tests.
+
+Having recently completed the pages for logged in users (account settings, manage projects, etc.) these pages seemed like the best candidates for our initial round of testing.
+
+The test I constructed was simple: asking users about their current use of PyPI, then requesting them to complete a series of tasks on the new management pages. For the test, I recruited users who were package owners or maintainers - i.e. people who had experienced the old user interface and who would understand my instructions.
+
+Over three weeks, I ran tests with six different users, with each test taking between 25 minutes and one hour; the duration depending on how much the test subject had to say.
+
+As expected, the tests highlighted a number of problems in the UI, which in turn prompted me to open or tag [20 issues](https://github.com/pypa/warehouse/issues?q=is%3Aissue+label%3A%22user+testing%22) in our issue tracker. The vast majority of these tickets are simple changes that have been tagged as [good first issues](https://github.com/pypa/warehouse/issues?q=is%3Aissue+label%3A%22good+first+issue%22) - which should in turn help with our efforts to outreach to new contributors.
+
+Several of the test participants also raised new ideas, for which I will open new tickets soon.
+
+A big thanks to Ernest W. Durbin III, Nicolas Dubois, Łukasz Langa, Russell Keith-Magee, Alex Chan and Eliot Berriot for participating in these tests.
+
+## Next Steps
+
+From this intial program it is clear that user testing can bring value to open source projects. The challenge is to continue to conduct such tests once our MOSS funding is depleted.
+
+Fortunately, the Python community is engaged with the Warehouse project, and, so far, we've had no problems recruiting test participants. Usability experts [recommend a minumum of 5 users per test](https://www.nngroup.com/articles/how-many-test-users/), so as long as this goodwill continues, I don't anticipate running out of test subjects.
+
+On the other hand, setting up and running user tests takes time; and it would be nice to have some help with this to lighten the workload (and increase our testing bus factor). For this reason, today I've put out a call to [train up to three people on how to conduct and run user tests](TODO). My hope is that this opportunity will be taken by students (or UX enthusiasts) as a way to build their experience by working on a live, widely used application.
```

Comment:
You want to link to https://gist.github.com/nlhkabu/a0b1ae0016a2641f6b79d9ace9110403 here, right? And for that you need a short explanation of PyPI?

Reply:
oh you got it :)